crates/dbs-tdx/src/tdx_ioctls.rs

```diff
@@ -0,0 +1,221 @@
+// Copyright © 2019 Intel Corporation
+//
+// Copyright (c) 2023 Alibaba Cloud.
+//
+// SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+//
+#![allow(missing_docs)]
+
+use std::os::unix::io::RawFd;
+
+use kvm_bindings::{CpuId, __IncompleteArrayField, KVMIO};
+use thiserror::Error;
+use vmm_sys_util::fam::{FamStruct, FamStructWrapper};
+use vmm_sys_util::ioctl::ioctl_with_val;
+use vmm_sys_util::{generate_fam_struct_impl, ioctl_ioc_nr, ioctl_iowr_nr};
+
+/// Tdx capability list.
+pub type TdxCaps = FamStructWrapper<TdxCapabilities>;
+
+/// Cpuid configs entry counts.
+const TDX1_MAX_NR_CPUID_CONFIGS: usize = 6;
+
+generate_fam_struct_impl!(
+    TdxCapabilities,
+    TdxCpuidConfig,
+    cpuid_configs,
+    u32,
+    nr_cpuid_configs,
+    TDX1_MAX_NR_CPUID_CONFIGS
+);
+
+#[repr(C)]
+#[derive(Debug, Default, Copy, Clone, PartialEq)]
+/// Tdx cpuid config.
+pub struct TdxCpuidConfig {
+    /// cpuid leaf
+    pub leaf: u32,
+    /// cpuid sub leaf
+    pub sub_leaf: u32,
+    /// eax
+    pub eax: u32,
+    /// ebx
+    pub ebx: u32,
+    /// ecx
+    pub ecx: u32,
+    /// edx
+    pub edx: u32,
+}
+
+#[repr(C)]
+#[derive(Default)]
+/// Tdx capabilities.
+pub struct TdxCapabilities {
```

Comment:
Add documention for  struct TdxCapabilities

Reply:
added.